src/main/java/com/dnd/reetplace/app/service/PlaceService.java

```diff
@@ -152,6 +165,47 @@ private List<PlaceGetResponse> updatePlaceIsBookmark(HttpServletRequest httpServ
         }).toList();
     }
 
+    /**
+     * (로그인 시) 장소 검색에서 각 장소 별 북마크 여부 및 북마크 id, 릿플점수를 업데이트한다.
+     * 또한, 카카오에서 제공하는 장소 정보 -> 앱 내에서 필요한 정보만 포함하는 Custom Response로 결과를 변환하여 반환한다.
+     * 이 때, 앱 내 카테고리로 분류되지 못하는 장소의 경우 결과에서 제외된다.
+     *
+     * @param httpServletRequest 로그인 여부를 판단하기 위한 HttpServletRequest 객체
+     * @param result             카카오 로컬 API를 통해 받아온 카카오에서 제공하는 장소 정보 List
+     * @return 북마크 여부, 북마크 id, 릿플점수가 업데이트 된 Custom Place Response List
+     */
+    private List<PlaceSearchResponse> updateSearchPlaceIsBookmark(HttpServletRequest httpServletRequest, List<KakaoPlaceSearchResponse> result) {
+        Member loginMember = findLoginMember(httpServletRequest);
+        List<PlaceSearchResponse> placeList;
+        if (loginMember == null) {
+            placeList = result.stream().map(place ->
+                    PlaceSearchResponse.of(place, null, null, null)
+            ).toList();
+        } else {
+            List<Bookmark> bookmarkList = bookmarkRepository.findAllByMember(loginMember.getId());
+            placeList = result.stream().map(place -> {
+                Optional<Bookmark> bookmark = bookmarkList.stream()
+                        .filter(b -> Objects.equals(b.getPlace().getKakaoPid(), place.getId()))
+                        .findFirst();
+                if (bookmark.isPresent()) {
+                    return PlaceSearchResponse.of(
+                            place,
+                            bookmark.get().getType(),
+                            bookmark.get().getId(),
+                            bookmark.get().getRate()
+                    );
+                } else {
+                    return PlaceSearchResponse.of(
+                            place,
+                            null,
+                            null,
+                            null);
+                }
+            }).toList();
+        }
+        return placeList.stream().filter(p -> p.getCategory() != null).toList();
```

Comment:
```suggestion
        if (loginMember == null) {
            return result.stream()
                    .map(PlaceSearchResponse::noBookmarkFrom)
                    .filter(place -> place.getCategory() != null)
                    .toList();
        }

        List<Bookmark> bookmarkList = bookmarkRepository.findAllByMember(loginMember.getId());
        return result.stream()
                .map(searchResponse -> bookmarkList.stream()
                        .filter(bookmark -> Objects.equals(bookmark.getPlace().getKakaoPid(), searchResponse.getId()))
                        .findFirst()
                        .map(bookmark -> PlaceSearchResponse.of(searchResponse, bookmark.getType(), bookmark.getId(), bookmark.getRate()))
                        .orElse(PlaceSearchResponse.noBookmarkFrom(searchResponse)))
                .filter(searchResponse -> searchResponse.getCategory() != null)
                .toList();
```

이 부분 이렇게 바꿔보면 어떨까요? 
로그인하지 않은 경우 바로 return하도록 하여 제어문의 depth를 줄였구요, 최대한 함수형 method들이 이어지도록 하여 불필요한 stream <-> list 변환 과정이 들어가지 않도록 했습니다.

Reply:
앗 참고로 저 코드를 구현하기 위해 `PlaceSearchResponse`에 다음과 같은 method도 추가했습니다.
```java
public static PlaceSearchResponse noBookmarkFrom(KakaoPlaceSearchResponse kakaoResponse) {
    return of(kakaoResponse, null, null, null);
}
```
북마크가 체크를 안하는 경우 일일히 `null`을 넣기보다는 이렇게 따로 method를 만들어주는게 사용성에서 좋을 것 같아서요.

Reply:
만약 이렇게 method 만들어서 쓰신다면 이름은 편하신대로 지으셔도 될 것 같습니다!
`of`, `from`, `ofWithoutBookmark`, ... 저도 그냥 급하게 한거라 막 지었어요

Reply:
depth 감소 및 Java Stream을 최대한 활용해서 코드 길이 및 가독성이 많이 좋아진 것 같습니다!
해당 코드로 수정하겠습니다!